sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -926,67 +1100,45 @@ static void send_telemetry_messages(void)
 
 static void temp_controller_build_telemetry_message(az_span payload, az_span* out_payload)
 {
+  int rc;
+
   int32_t working_set_ram_in_kibibytes = rand() % 128;
 
-  pnp_build_telemetry_message(
-      payload,
-      telemetry_working_set_name,
-      append_int32_callback,
-      (void*)&working_set_ram_in_kibibytes,
-      out_payload);
+  az_json_writer jr;
+
+  const char* const log = "Failed to build telemetry payload";
+
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc = az_json_writer_init(&jr, payload, NULL), log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc = az_json_writer_append_begin_object(&jr), log);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
```

Comment:
I think you already updated this in a later PR to not use an rc here?

Reply:
yup 👍